feat: 내 명함 위젯에게 새로운 timeline 요청 (#515)

hyun99999 opened the pull request:
## 🌴 PR 요약

🌱 작업한 브랜치
- #515

🌱 작업한 내용
- 로그인, 로그아웃, 회원탈퇴, 해당명함 삭제 시 위젯의 timeline 을 요청하도록 구현하였습니당
    - 회원탈퇴 or 로그아웃할 때는 위젯을 엠티뷰로 업데이트하고,
    - 명함이 삭제될 때는 위젯을 대표명함(첫 번째 명함)으로 업데이트하겠습니다.

## 📸 스크린샷
|기능|스크린샷|
|:--:|:--:|
|로그아웃|<img src="https://github.com/TeamNADA/NADA-iOS-ForRelease/assets/69136340/52206886-bc14-4212-9650-b7aa9f17d2df" width ="200">|
|해당 명함 삭제|<img src="https://github.com/TeamNADA/NADA-iOS-ForRelease/assets/69136340/89dbd5cd-abfe-4bef-9fb0-66c880c83efd" width ="200">|

## 📮 관련 이슈
- Resolved: #515

hyun99999 committed:
chore: AppGroup 을 WidgetExtension 에 추가 (#515)

hyun99999 committed:
chore: TargetMembership 수정 (#515)

- GenericResponse 은 서버통신할 때 데이터 모델을 위해서
- UserDefaults+Extension 은 액세스 토큰을 app group 을 사용하여 공유하기 위해서

hyun99999 committed:
feat: WidgetKit 에게 새로운 timeline 요청 (#515)

- CardListViewController 에서 명함 삭제 시

hyun99999 committed:
feat: WidgetKit 에게 새로운 timeline 요청 (#515)

- LoginViewController 에서 로그인 시

hyun99999 committed:
feat: WidgetKit 에게 새로운 timeline 요청 (#515)

- MoreViewController 에서 로그아웃, 회원탈퇴 시

hyun99999 committed:
chore: pod update (#515)

- cocoapods version 1.12.1 최신반영

hyun99999 committed:
feat: widget 내에서 서버통신 구현 (#515)

hyun99999 committed:
feat: getTimeline 메서드 수정 (#515)

hyun99999 commented:
#512 PR부터 코리하시공 머지 된 다음에 코드 체인지드 된 파일 봐주시면 편하실거에요!

hyun99999 commented:
코리 후에 주석 삭제하겠습니당😃

dlwns33 commented on Widgets/WidgetsBundle/MyCardWidget.swift:
```diff
@@ -34,29 +34,58 @@ struct MyCardProvider: IntentTimelineProvider {
+                    }
+                case .failure(let error):
+                    print(error)
+                    // 회원 탈퇴, 로그아웃
+                    let entry = MyCardEntry(date: entryDate, widgetCard: nil)
+                    entries = [entry]
+                    let timeline = Timeline(entries: entries, policy: .atEnd)
+                    completion(timeline)
```
로그아웃 분기처리 확인했습니당

hyun99999 replied:
`ITMS-90626: Invalid Siri Support - Localized title for custom intent: 'MyCard' not found for locale: ko_KR`
의 이슈를 고쳐달라고 했는데요. 현재 앱이 ko_KR 로 기본 언어가 설정되어 있는데 custom intent 의 title 이 영어라서 로컬라이즈되지 않았다고 하는 것 같습니다! 그래서 다음과 같이 수정해서 커밋하였습니당

<img width="600" alt="스크린샷 2023-05-12 오후 4 20 07" src="https://github.com/TeamNADA/NADA-iOS-ForRelease/assets/69136340/bd5b9be6-f6e5-4ab4-b950-371632288c4c">

dlwns33 reviewed (approved):
수고하셨씁니당! 혹시 그 저희 자꾸 나오는 에러 관련해서 한번 봐주실 수 있나용? 로컬라이제이션 어쩌구..

hyun99999 committed:
refactor: IntentHandler 가 가진 cardItems 프로퍼티 삭제 (#515)

hyun99999 committed:
chore: MyCard custom intent title 수정 (#515)

- ITMS-90626: Invalid Siri Support - Localized title for custom intent: 'MyCard' not found for locale: ko_KR 해결

hyun99999 committed:
Merge branch 'develop' of https://github.com/TeamNADA/NADA-iOS-ForRelease into feature/#515
